### PR TITLE
feat: add Describe a resource

### PR DESCRIPTION
**Pull request by MdechampG**
## Summary by CodeRabbit

- **Documentation**
    - Updated the comprehensive guide on describing and registering digital resources in the OKP4 protocol, emphasizing secure key management, resource descriptions, and blockchain registration.
    - Added a guide on establishing resource governance within the OKP4 Protocol's dataverse, covering Governance VPs, rules definition, Verifiable Credentials, and resource management.
    - Introduced functionality for creating and managing governance within specific zones in the Dataverse, focusing on self-governance, Prolog programs, Governance VPs, and blockchain registration.

### CODE REVIEW

File: docs/academy/describe-resource.mdx [hunk 1]
```diff
@@ -0,0 +1,275 @@
...
+The mnemonic serves as a backup mechanism for the wallet. You can regenerate the key pair by inputting the mnemonic into any compatible wallet software to recover access to your funds.
+
+Here, we don’t create a wallet for cryptocurrency storage but for secure storage of identifiers: A wallet can securely store digital identifiers, such as proofs of identity, certificates, or other personal identification information. 
+
+:::danger
+A keyring is a secure software utility designed to store and manage credentials, such as passwords, cryptographic keys, and API tokens, in a centralized and encrypted form. `--keyring-backend test` is a command used to configure a keyring in test mode, which is helpful for development and testing but not for applications where private key security is a major concern. Be careful; this keyring is unsafe as the private keys are unencrypted on the file system.
```
Comment (amimart) [inline docs/academy/describe-resource.mdx:40]:
```suggestion
A keyring is a secure software utility designed to store and manage credentials, such as passwords, cryptographic keys, and API tokens, in a centralized and encrypted form. `--keyring-backend test` is a command parameter used to configure a keyring in test mode, which is helpful for development and testing but not for applications where private key security is a major concern. Be careful; this keyring is unsafe as the private keys are unencrypted on the file system.
```

File: docs/academy/describe-resource.mdx [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+- `hasTitle`: fill in the title of the service
+- `hasWebpage`: fill in the URL of the service’s webpage
+- `id` (issuer): copy the did:key of the issuer
+- `name`: put the issuer's name (string). 
+
+## Step 3: Sign the credentials
+
+Now that you have created the VCs, you will sign them. 
+
+Signing a verifiable credential involves creating a digital signature using cryptographic techniques. This signature is unique to both the document (in this case, the credential) and the signer's private key, making it nearly impossible to forge. The private key is kept secret by the signer, while the corresponding public key is made available for anyone wishing to verify the signature's authenticity.
```
Comment (amimart) [inline docs/academy/describe-resource.mdx:235]:
```suggestion
Signing a verifiable credential involves creating a digital signature using cryptographic techniques. This signature is unique to both the document (in this case, the credential) and the signer's private key, making it impossible to forge. The private key is kept secret by the signer, while the corresponding public key is made available for anyone wishing to verify the signature's authenticity.
```

**Review by amimart — APPROVED** [review]
Great work that's pretty nice! 🙏 

I just noted some suggestions :)

File: docs/academy/describe-zone.mdx [hunk 3]
```diff
@@ -0,0 +1,204 @@
+---
+sidebar_position: 12
+sidebar_class_name: hidden
+---
+
+# Describe a Zone
+
+<i>Reading time: {readingTime} min</i>
```
[the remark this answers is not in the document]
Comment (MdechampG) [reply docs/academy/describe-zone.mdx:8]: It is a variable.

File: docs/academy/resource-governance.mdx [hunk 4]
```diff
@@ -0,0 +1,338 @@
+---
+sidebar_position: 11
+sidebar_class_name: hidden
+---
+
+# Create the resource's governance
+
+<i>Reading time: {readingTime} min</i>
```
[the remark this answers is not in the document]
Comment (MdechampG) [reply docs/academy/resource-governance.mdx:8]: It is a dynamic variable.

File: docs/academy/zone-governance.mdx [hunk 5]
```diff
@@ -0,0 +1,311 @@
...
+
+Governance rules of the Collaborative AI zone: 
+
+| Division    | Ordinal number | Title                                                |
+|-------------|----------------|------------------------------------------------------|
+| Chapter     | 1              | Collaborative AI zone                                |
+| Section     | 1.1            | Zone Governance                                      |
+| Sub-section | 1.1.1          | Governers                                            |
+| Article     | 1.1.1.1        | Conditions to amend the rules                       |
+| Paragraph   | 1.1.1.1.1      | Governance can only be amended by the identity `did:key:0x04d1j1x8f8a7a28f9a5a254c326a963a22f5a5b5d5f5e5d5c5b5a5958575758`|
```
Comment (amimart) [inline docs/academy/zone-governance.mdx:60]:
```suggestion
| Paragraph   | 1.1.1.1.1      | Governance can only be amended by the identity `did:key:zQ3shs7auhJSmVJpiUbQWco6bxxEhSqWnVEPvaBHBRvBKw6Q3`|
```

File: docs/academy/zone-governance.mdx [hunk 6]
```diff
@@ -0,0 +1,311 @@
...
+         title('sec1.1', 'Zone Governance').
+         subSection('subSec1.1.1').
+             partOf('subSec1.1.1', 'sec1.1').
+             title('subSec1.1.1', 'Governers').
+             article('art1.1.1.1').
+                 partOf('art1.1.1.1', 'subSec1.1.1').
+                 title('art1.1.1.1', 'Conditions to amend the rules').
+                 paragraph('para1.1.1.1.1', permitted) :- action(A), A=='governance:amend', S == 'did:key:0x04d1j1x8f8a7a28f9a5a254c326a963a22f5a5b5d5f5e5d5c5b5a5958575758'.
+                    partOf('para1.1.1.1.1', 'art1.1.1.1').
+                     description('para1.1.1.1.1', 'Governance can only be amended by the identity `did:key:0x04d1j1x8f8a7a28f9a5a254c326a963a22f5a5b5d5f5e5d5c5b5a5958575758`').
```
Comment (amimart) [inline docs/academy/zone-governance.mdx:206]:
```suggestion
                     description('para1.1.1.1.1', 'Governance can only be amended by the identity `did:key:zQ3shs7auhJSmVJpiUbQWco6bxxEhSqWnVEPvaBHBRvBKw6Q3`').
```

File: docs/academy/zone-governance.mdx [hunk 7]
```diff
@@ -0,0 +1,311 @@
...
+                        "type": ":SubSection",
+                        "hasTitle": "Governers",
+                        "hasOrdinalNumber": "1.1.1",
+                        "hasArticle": {
+                            "type": ":Article",
+                            "hasTitle": "Conditions to amend the rules",
+                            "hasOrdinalNumber": "1.1.1.1",
+                            "hasParagraph": {
+                                "type": ":Paragraph",
+                                "hasTitle": "Governance can only be amended by the identity `did:key:0x04d1j1x8f8a7a28f9a5a254c326a963a22f5a5b5d5f5e5d5c5b5a5958575758`",
```
Comment (amimart) [inline docs/academy/zone-governance.mdx:262]:
```suggestion
                                "hasTitle": "Governance can only be amended by the identity `did:key:zQ3shs7auhJSmVJpiUbQWco6bxxEhSqWnVEPvaBHBRvBKw6Q3`",
```

File: docs/academy/describe-resource.mdx [hunk 8]
```diff
@@ -0,0 +1,275 @@
...
+| Title | IPFS | 
+|--------------|------|
+| Description | IPFS is a decentralized protocol designed to facilitate the storage, sharing, and retrieval of files on a global scale. | 
+| Category | Storage | 
+| Image | NA |
+| Publisher | OKP4 |
+| Tags | Storage, Cloud, Peer-to-peer |
+| Webpage | https://docs.ipfs.tech/ |
+
+```bash
```
Comment (ccamel) [inline docs/academy/describe-resource.mdx:188]:
```suggestion
```json
```

File: docs/academy/describe-zone.mdx [hunk 9]
```diff
@@ -0,0 +1,204 @@
...
+| Title | Collaborative AI Zone | 
+|--------------|------|
+| Description | Collaborative AI Zone is dedicated to training AI models in a collaborative and open way. | 
+| Image | NA |
+| Tags | AI, Collaboration, Machine learning |
+| Topic | Other |
+
+Let’s fill in the template.
+
+```bash
```
Comment (ccamel) [inline docs/academy/describe-zone.mdx:118]:
```suggestion
```json
```

File: docs/academy/resource-governance.mdx [hunk 10]
```diff
@@ -0,0 +1,337 @@
...
+
+Governance VPs play an essential role in the dataverse, applying universally to various categories of resources, such as datasets, services, and [zones](/academy/zone-governance).
+
+Instantiate the template [credential-governance-text](https://github.com/okp4/ontology/blob/main/src/example/governance/crime-dataset-governance.jsonld). 
+
+Fill in the template with the elements you [precedently defined](#step-1-define-the-governance-of-your-resource). 
+
+The following code specifies that the Crime Dataset whose DID is `<did:example:ebfeb1f712ebc6f1c276e12ec21>` has a governance system that is described by the prolog program `<cosmwasm:okp4-objectarium:okp41lppz4x9dtmccek2m6cezjlwwzup6pdqrkvxjpk95806c3dewgrfq602kgx...>` for which a text description is also provided.
+
+```bash
```
Comment (ccamel) [inline docs/academy/resource-governance.mdx:252]:
```suggestion
```json
```

File: docs/academy/zone-governance.mdx [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+
+Governance VPs play an essential role in the dataverse, applying universally to various categories of resources, such as [datasets, services](/academy/resource-governance), and zones.
+
+Instantiate the template [credential-governance-text](https://github.com/okp4/ontology/blob/main/src/example/governance/collab-AI-zone-governance.jsonld). 
+
+Fill in the template with the elements you [precedently defined](#step-1-define-the-governance-of-your-zone). 
+
+The following code specifies that the Collaborative AI zone whose DID is `<did:key:zQ3shunN7us5oKNYKKzprW8nP5HAoTf4t2Fxyudho5zMAR1PQ>` has a governance system that is described by the prolog program `<cosmwasm:okp4-objectarium:okp41lppz4x9dtmccek2m6cezjlwwzup6pdqrkvxjpk95806c3dewgrfq602kgx...>` for which a text description is also provided.
+
+```bash
```
Comment (ccamel) [inline docs/academy/zone-governance.mdx:230]:
```suggestion
```json
```

File: docs/academy/describe-zone.mdx [hunk 12]
```diff
@@ -0,0 +1,204 @@
...
+
+Here, we don’t create a wallet for cryptocurrency storage but for secure storage of identifiers: A wallet can securely store digital identifiers, such as proofs of identity, certificates, or other personal identification information. 
+
+:::danger
+A keyring is a secure software utility designed to store and manage credentials, such as passwords, cryptographic keys, and API tokens, in a centralized and encrypted form. `--keyring-backend test` is a command used to configure a keyring in test mode, which is helpful for development and testing but not for applications where private key security is a major concern. Be careful; this keyring is unsafe as the private keys are unencrypted on the file system.
+:::
+
+```bash
+​​# Create a new key. Replace "collab-ai-zone" with another name
+okp4d keys add collab-ai-zone --keyring-backend test
```
[the remark this answers is not in the document]
Comment (MdechampG) [reply docs/academy/describe-zone.mdx:42]: We will clarify later.

File: docs/academy/describe-zone.mdx [hunk 13]
```diff
@@ -0,0 +1,204 @@
...
+
+COMMANDE A AJOUTER
+
+You can see that there is a new field in the jsonld with the cryptographic proof. 
+
+AJOUTER RESULTAT
+
+The VC is now in the hands of the Holder. Note that it is possible that the Issuer is also the Holder. 
+
+The OKP4 blockchain can only register VCs in N-Quads format. Then, you must convert the jsonld files in N-Quads. You can use this tool: https://transform.tools/jsonld-to-nquads .
```
[the remark this answers is not in the document]
Comment (MdechampG) [reply docs/academy/describe-zone.mdx:178]: Thank you we will add it in a further PR.

File: docs/academy/describe-resource.mdx [hunk 14]
```diff
@@ -0,0 +1,275 @@
+---
+sidebar_position: 10
+sidebar_class_name: hidden
+---
+
+# Describe a resource
+
+<i>Reading time: {readingTime} min</i>
```
[the remark this answers is not in the document]
Comment (MdechampG) [reply docs/academy/describe-resource.mdx:8]: It's a dynamic variable.

**Review by ccamel — APPROVED** [review]
Great piece ! Thanks 👍

**Pull request merged**